### ISSUE TITLE
azurerm_data_factory_linked_service_sql_server linked SQL server issue AzureSQL

### ISSUE THREAD

**Issue by kylesgarage** [terraform-providers/terraform-provider-azurerm#6417]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
Terraform v0.12.24
+ provider.acme v1.5.0
+ provider.azuread v0.8.0
+ provider.azurerm v2.4.0
+ provider.cloudflare v2.5.1
+ provider.random v2.2.1
+ provider.tls v2.1.1
```

### Affected Resource(s)

* `azurerm_data_factory_linked_service_sql_server`

### Terraform Configuration Files

```hcl
resource "azurerm_data_factory_linked_service_sql_server" "sql_linked" {
    name                = "${local.prefix}-datafactory-ls-db"
    resource_group_name = azurerm_resource_group.backendrg.name
    data_factory_name   = azurerm_data_factory.backenddatafactory.name
    connection_string   = "Integrated Security=False;Encrypt=True;Connection Timeout=30;Data Source=${var.sql_server_name};${var.db_conn_string_managed_id}"
    additional_properties = {
        type = "AzureSqlDatabase"
    }
}
```
### Expected Behavior

Running apply should create the resource without error

### Actual Behavior

Received ab error about unable to classify the Data Factory Linked Service SQL Server

Error: Error classifiying Data Factory Linked Service SQL Server "backend-datafactory-ls-sql" (Data Factory "backend-datafactory" / Resource Group "backend-rg"): Expected: "SqlServer" Received: "Microsoft.DataFactory/factories/linkedservices"

  on ..\modules\backend\datafactory.tf line 11, in resource "azurerm_data_factory_linked_service_sql_server" "sql_inked":
  11: resource "azurerm_data_factory_linked_service_sql_server" "sql_linked" {

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Tried in AzureChinaCloud and AzureCloud, both the same response.
It may be worth noting this is to create a linked AzureSQL (hence the additional_properties), not the standard SQL Server. 
The attribute **does** create, despite the error.

* #0000

**Comment by yupwei68** [comment]
Hi @kylesgarage , thanks for opening this issue. Actually, data_factory_linked_service for AzureSqlDatabase has not been supported. I have tried to remove the following, and it works for me.
```
additional_properties = {
        type = "AzureSqlDatabase"
    }
```

My code:
```
resource "azurerm_data_factory_linked_service_sql_server" "sql_linked" {
    name                = "datafactory-ls-db"
    resource_group_name = "yupserver"
    data_factory_name   = "yupfac"
    connection_string   = "Integrated Security=False;Encrypt=True;Connection Timeout=30;Data Source=yupserver;User ID=yup;Password=secret"
}
```

**Comment by katbyte** [comment]
Hi @kylesgarage,

When using the specific `azurerm_data_factory_linked_service_sql_server` resources the type property is set internally by the terraform resource thus i don't think you need to include it in the additional properties. Looking at our nightly acceptance tests the following config is working:

```
resource "azurerm_data_factory_linked_service_sql_server" "test" {
  name                = "acctestlssql%d"
  resource_group_name = azurerm_resource_group.test.name
  data_factory_name   = azurerm_data_factory.test.name
  connection_string   = "Integrated Security=False;Data Source=test;Initial Catalog=test;User ID=test;Password=test"
  annotations         = ["test1", "test2", "test3"]
  description         = "test description"

  parameters = {
    foo = "test1"
    bar = "test2"
  }

  additional_properties = {
    foo = "test1"
    bar = "test2"
  }
}
```

Could you try that removing the type line and let us know if you are still having difficulties?

**Comment by mbfrahry** [comment]
After a bit of digging, it looks like this warrants a new resource. I'll start actively working on getting that ready!

**Comment by mbfrahry** [comment]
This resource was just merged in #8349 and should go out in the next release.
